### PR TITLE
Runtime: small refactoring in parsing.js

### PR DESCRIPTION
**Pull request by hhugo**
spotted while working on https://github.com/ocsigen/js_of_ocaml/pull/1672

cc @smorimoto

### CODE REVIEW

File: runtime/parsing.js [hunk 1]
```diff
@@ -129,8 +129,8 @@ function caml_parse_engine(tables, env, cmd, arg) {
   var state = env[env_state];
   var errflag = env[env_errflag];
 
-  exit: for (;;) {
-    next: switch (cmd) {
+  the_loop: for (;;) {
```
Comment (smorimoto) [inline runtime/parsing.js:132]: Is that really needed?

```suggestion
  for (;;) {
```

File: runtime/parsing.js [hunk 2]
```diff
@@ -147,7 +147,7 @@ function caml_parse_engine(tables, env, cmd, arg) {
           break;
         }
         res = READ_TOKEN;
-        break exit;
+        break the_loop;
```
Comment (smorimoto) [inline runtime/parsing.js:150]:
```suggestion
        break;
```

File: runtime/parsing.js [hunk 3]
```diff
@@ -206,7 +206,7 @@ function caml_parse_engine(tables, env, cmd, arg) {
             ) {
               if (caml_parser_trace) log("Recovering in state " + state1);
               cmd = shift_recover;
-              break next;
+              continue the_loop;
```
Comment (smorimoto) [inline runtime/parsing.js:209]:
```suggestion
              continue;
```

File: runtime/parsing.js (hunk 1 above)
Comment (smorimoto) [reply runtime/parsing.js:132]: nvm

**Review by smorimoto — APPROVED** [review]

File: runtime/parsing.js (hunk 2 above)
Comment (vouillon) [reply runtime/parsing.js:150]: You want to exit the loop and not just terminate the current switch statement.

Comment (smorimoto) [reply runtime/parsing.js:150]: Indeed

**Pull request merged**